rdrf/rdrf/forms/dsl/code_evaluator.py

```diff
@@ -0,0 +1,149 @@
+from datetime import datetime
+from itertools import chain
+import logging
+
+from ..dynamic.value_fetcher import DynamicValueFetcher
+from .constants import VISIBLE, HIDDEN
+from .parse_operations import BooleanOp, Condition, parse_dsl, transform_tree
+from .parse_utils import CDEHelper, SectionHelper
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConditionEvaluator:
+
+    INTEGER_TYPE = 'integer'
+    DATE_TYPE = 'date'
+
+    def __init__(self, operator, value, existing_value, data_type):
+        self.operator = operator
+        self.value = value
+        self.existing_value = existing_value
+        self.data_type = data_type
+
+    @staticmethod
+    def parse_date(value):
+        try:
+            return datetime.strptime(value, '%d-%m-%Y')
+        except ValueError:
+            try:
+                return datetime.strptime(value, '%Y-%m-%d')
+            except ValueError:
+                logger.error(f"Cannot parse date: {value}")
+                return None
+
+    @staticmethod
+    def parse_int(value):
+        try:
+            return int(value)
+        except ValueError:
+            logger.error(f"Cannot parse int: {value}")
+            return 0
```

Comment:
Why doesn't this return `None`, just like parse date?
I would prefer having it consistent, ie. if we can't parse a value we return `None`.

Reply:
Updated.